drivers/i2c/busses/i2c-bcm2708.c

```diff
@@ -160,7 +160,7 @@ static inline int bcm2708_bsc_setup(struct bcm2708_i2c *bi)
 			bcm2708_wr(bi, BSC_C, BSC_C_CLEAR_1);
 
 			/* Fill FIFO with entire write message (16 byte FIFO) */
-			while (bi->pos < bi->msg->len) {
+			while ((bcm2708_rd(bi, BSC_S) & BSC_S_TXD) && (bi->pos < bi->msg->len)) {
```

Comment:
I agree. For the same reason I'll also adjust the while loop in bcm2708_bsc_fifo_fill.
I'll create a new pull request.

Thx for your review.